### PR TITLE
EnhancediOSProxyInterface

### PR DESCRIPTION
**Pull request by Toyota-BSalahat**
Currently when app developers implement SDL they are responsible for understanding and handling a lot of SDL-centric logic. This creates a heavy burden for app implementers, leads to large bug/issue counts in production integrations, and results in a high cost of development/maintenance to support SDL. This proposal aims to enhance the proxy defined interface to SDL to reduce the burden on app developers, centralize SDL-specific logic within the proxy where it can be maintained by developers who understand SDL-specific implementation details, and reduce the time and cost required to integrate with SDL successfully.

### CODE REVIEW

**Comment by Toyota-BSalahat** [comment]
@theresalech this should be review ready.

File: proposals/NNNN-EnhancediOSProxyInterface.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+Currently app integrations must handle a lot of tedious logic to manage show requests, manage SDL application state, and in general manage their application flow within SDL.
+This leads to a large burden on 3rd party app developers, and requires a lot of custom code to be written specifically for SDL. The current interface for SDL is also dissimilar from the design concepts encouraged by iOS, and that iOS developers are familiar with.
+
+Native iOS development is heavily tied to the concept of Model View Controller (MVC), and organizing your app flow via View Controllers and the view stack.
+
+![MVC](https://developer.apple.com/library/content/documentation/General/Conceptual/DevPedia-CocoaCore/Art/model_view_controller_2x.png "MVC")
+[Apple Documentation](https://developer.apple.com/library/content/documentation/General/Conceptual/DevPedia-CocoaCore/MVC.html)
+
+These view controller's often conform to delegation interfaces which allow an eloquent way of populating their view components with content.
```
Comment (theresalech) [inline proposals/NNNN-EnhancediOSProxyInterface.md]: Please update to "view controllers" (removing apostrophe).

File: proposals/NNNN-EnhancediOSProxyInterface.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+![Delegation](https://developer.apple.com/library/content/documentation/General/Conceptual/DevPedia-CocoaCore/Art/delegation_2x.png "Delegation")
+[Apple Documentation](https://developer.apple.com/library/content/documentation/General/Conceptual/DevPedia-CocoaCore/Delegation.html)
+
+
+However, the current SDL proxy interface is designed at a lower level than this. It does not currently mesh well with these higher level design patterns. This often results in app integrations having to write completely custom SDL modules of code within their code base.
+
+![App Flow](https://i.imgur.com/f4kekYV.jpg "App Flow")
```
Comment (theresalech) [inline proposals/NNNN-EnhancediOSProxyInterface.md]: Please upload image to /assets/proposals.  You can reference [0084](https://github.com/smartdevicelink/sdl_evolution/issues/250) as an example for embedding images.

File: proposals/NNNN-EnhancediOSProxyInterface.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+2. The implementation complexity of managing SDL's display state and other fine details will be taken into the proxy, this will probably increase maintenance for the project maintainer/open source community, but will take that burden from future app implementations
+
+## Impact on existing code
+
+Technically impact to existing code would be minimal, however this would be a large addition of new code.
+
+## Alternatives considered
+
+1. Continue requiring apps to understand all SDL implementation details, and continue to not have a smooth integration experience.
+3. Implement code that manages some of the trickier SDL logic, but still requires app developers to develop independent SDL code modules that don't really conform to MVC or other familiar iOS patterns [268](https://github.com/smartdevicelink/sdl_evolution/pull/268)
```
Comment (theresalech) [inline proposals/NNNN-EnhancediOSProxyInterface.md]: I think this should be +2.

**Comment by Toyota-BSalahat** [comment]
@theresalech files should be updated now.

**Pull request merged**